Samples/RedundantHubSample/RedundantHubSample/RedundantNotificationHubClient.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.NotificationHubs;
+
+namespace RedundantHubSample
+{
+    public enum DefaultNamespace
+    {
+        Primary = 1,
+        Backup = 2
+    }
+
+    public class RedundantNotificationHubClient
+    {
+        private readonly INotificationHubClient _primaryNotificationHubClient;
+        private readonly INotificationHubClient _backupNotificationHubClient;
+
+        public RedundantNotificationHubClient(string primaryConnectionString, string backupConnectionString, string hubName)
+        {
+            _primaryNotificationHubClient = NotificationHubClient.CreateClientFromConnectionString(primaryConnectionString, hubName);
+            _backupNotificationHubClient = NotificationHubClient.CreateClientFromConnectionString(backupConnectionString, hubName);
+        }
+
+        public DefaultNamespace DefaultNamespace { get; set; }
+
+        public Task CreateOrUpdateInstallationAsync(Installation installation, CancellationToken cancellationToken = default)
+            => Task.WhenAll(
+                _primaryNotificationHubClient.CreateOrUpdateInstallationAsync(installation, cancellationToken),
+                _backupNotificationHubClient.CreateOrUpdateInstallationAsync(installation, cancellationToken));
+
+        public async Task<Installation> GetInstallationAsync(string installationId)
+        {
+            while (true)
```

Comment:
This while loop and try/catch don’t belong here. Please move it out to the method calling it.